source/guides/kernel/firmware.rst

```diff
@@ -0,0 +1,77 @@
+.. _firmware:
+
+Firmware
+########
+
+This guide shows how |CL-ATTR| handles firmware or microcode loading.
+
+.. contents::
+   :local:
+   :depth: 1
+
+Overview 
+********
+
+Many devices and system components require firmware or microcode, software
+that runs directly on the device, to function correctly. Because firmware
+loading requires privileged hardware access, the kernel is involved in the
+process.
+
+Firmware does not typically come with source code. Instead, firmware is
+provided as binary blobs which are licensed for free or non-free use.
+
+In |CL| firmware is loaded during device initialization which typically
+happens at boot time. 
+
+
+Included firmware
+*****************
+
+The Linux kernel sources contain firmware binaries that are licensed to allow
+free redistribution. |CL| packages these firmware binaries in the
+`linux-firmware bundles
+<https://clearlinux.org/software?search_api_fulltext=linux-firmware>`_ and
+automatically includes them with the kernel.
+
+The Linux kernel's firmware repository can be found here:
+https://git.kernel.org/pub/scm/linux/kernel/git/firmware/linux-firmware.git
+
+If the firmware for your device is included upstream, no action is required
+for it to be automatically detected and loaded. 
+
+
+Loading additional firmware
+***************************
+
+Some device hardware manufactures have a license that limits redistribution of
+firmware which means |CL| is unable to distribute those firmware and you must
+manually obtain them from the manufacture or another source.
+
+You can place additional firmware in :file:`/etc/firmware`. |CL| reads this
+directory for additional firmware files in conjunction with the typical
+:file:`/lib/firmware` path to provide a :ref:`stateless design <stateless>`.
+
+
+#. Create the :file:`/etc/firmware` directory
+
+   .. code-block:: bash
+
+      sudo mkdir -p /etc/firmware
+
+#. Obtain the additional firmware binary from a trusted source.
+
+#. Copy the firmware files including any subdirectories to
+   :file:`/etc/firmware`. It is important to place the firmware files in
+   expected path for proper loading. 
+
+   .. code-block:: bash
+
+      sudo cp -Rv <directory>/<filename>.<fw|bin> /etc/firmware 
+
+
```

Comment:
You might consider an example here, such as something from https://www.intel.com/content/www/us/en/support/articles/000005511/network-and-i-o/wireless-networking.html

Reply:
...of course pointing out that it's already provided in /usr/lib/firmware by the `linux-firmware-wifi` bundle.

Reply:
If we're going to add a real example, I rather find an example that isn't an exception/already included and ideally one that has a semi-official source :/.

Reply:
https://storage.microsemi.com/en-us/speed/scsi/linux/aic94xx-seq-30-1_tar_gz.php

Reply:
This is where I'm trying to look:
https://wiki.debian.org/Firmware

But for the most part, it looks like if it's available from an official source, we already have it.

Reply:
I'm going to leave this generic for now. If someone from the community surfaces a good example, we can add it later.